### PR TITLE
Breakout rest of the enums from dCommonVars and clean it up

### PR DESCRIPTION
**Pull request by aronwk-aaron**
so we don't have to deal with merge conflicts
ePlayerFlags is not a scoped enum, yet, due to it's complexity and varied usage

### CODE REVIEW

**Comment by aronwk-aaron** [comment]
Played for about an hour or so to test the areas that were touched.
These are all non-functional changes.

File: dGame/dComponents/PropertyManagementComponent.cpp
Comment (aronwk-aaron) [inline dGame/dComponents/PropertyManagementComponent.cpp:501]: this was the only "functional" change, since it was an error for it to be like this in the first place

File: dGame/dComponents/PropertyManagementComponent.cpp
Comment (aronwk-aaron) [inline dGame/dComponents/PropertyManagementComponent.cpp:554]: as well as here

File: dGame/dComponents/PropertyManagementComponent.cpp
Comment (EmosewaMC) [reply dGame/dComponents/PropertyManagementComponent.cpp:501]: I care more about why it's an error than it being an error

Comment (aronwk-aaron) [reply dGame/dComponents/PropertyManagementComponent.cpp:501]: Smash takes (LWOOBJID, ekilltype), not just (ekilltype) as it was previously, it was before having undefined behavior

File: dGame/dComponents/PetComponent.cpp
Comment (EmosewaMC) [inline dGame/dComponents/PetComponent.cpp:887]: I will note, the specific description of this flag as we have it is "First manual pet hibernate" and not "Activated Pet".   This may be confusing to future maintainers having a mis-match

File: dGame/dGameMessages/GameMessages.cpp
Comment (EmosewaMC) [inline dGame/dGameMessages/GameMessages.cpp:5149]: I will note, flags are signed ints, and not unsigned ints.

File: dScripts/02_server/Map/AM/AmBlueX.cpp
Comment (EmosewaMC) [inline dScripts/02_server/Map/AM/AmBlueX.cpp:21]: Can you confirm this still works with the above knowledge that flags are ints and not unsigned ints?

File: dGame/dComponents/PetComponent.cpp
Comment (aronwk-aaron) [reply dGame/dComponents/PetComponent.cpp:887]: will fix

File: dGame/dGameMessages/GameMessages.cpp
Comment (aronwk-aaron) [reply dGame/dGameMessages/GameMessages.cpp:5149]: I thought the underlying type for int was uint32_t, my mistake

File: dGame/dComponents/PetComponent.cpp
Comment (aronwk-aaron) [reply dGame/dComponents/PetComponent.cpp:887]: Also, I want to eventually go through and defined all of these that are just magic numbers scattered throughout the codebase. but that was a larger undertaking that I decided not to do in this PR. Could you point me in the direction of where we have those flags?

File: dScripts/02_server/Map/Property/AG_Small/ZoneAgProperty.cpp
Comment (EmosewaMC) [inline dScripts/02_server/Map/Property/AG_Small/ZoneAgProperty.cpp:416]: If we're changing these to int32_t in this PR their setter in each respective property setter needs to be updated (right now these are stored as a uint32 since at the time we marked them as unsigned)

**Review by DarwinAnim8or — DISMISSED** [review]
this commit is so large i think it gave me whiplash

**Comment by aronwk-aaron** [comment]
> this commit is so large i think it gave me whiplash

Just wait, you'll have to approve it again after I address emo's review

**Review by DarwinAnim8or — APPROVED** [review]

**Review by EmosewaMC — APPROVED** [review]
large approval

**Pull request merged**
